gui/guihtml/src/TGHtmlUri.cxx

```diff
@@ -203,51 +203,28 @@ int TGHtmlUri::ComponentLength(const char *z, const char *zInit, const char *zTe
 
 char *TGHtmlUri::BuildUri()
 {
-   int n = 1;
-   char *z;
-
-   if (fZScheme)    n += strlen(fZScheme) + 1;
-   if (fZAuthority) n += strlen(fZAuthority) + 3;
-   if (fZPath)      n += strlen(fZPath) + 1;
-   if (fZQuery)     n += strlen(fZQuery) + 1;
-   if (fZFragment)  n += strlen(fZFragment) + 1;
-   z = new char[n];
-   if (z == 0) return 0;
-   n = 0;
-   if (fZScheme) {
-      // coverity[secure_coding]
-      sprintf(z, "%s:", fZScheme);
-      n = strlen(z);
-   }
-   if (fZAuthority) {
-      // coverity[secure_coding]
-      sprintf(&z[n], "//%s", fZAuthority);
-      n += strlen(&z[n]);
-   }
+   TString uri;
+
+   if (fZScheme)
+      uri.Append(Form("%s:", fZScheme));
+
+   if (fZAuthority)
+      uri.Append(Form("//%s", fZAuthority));
+
    if (fZAuthority && fZAuthority[strlen(fZAuthority)-1] != '/' &&
-      !(fZPath && fZPath[0] == '/')) {
-      // coverity[secure_coding]
-      strcat(z, "/"); // NOLINT
-      ++n;
-   }
-   if (fZPath) {
-      // coverity[secure_coding]
-      sprintf(&z[n], "%s", fZPath);
-      n += strlen(&z[n]);
-   }
-   if (fZQuery) {
-      // coverity[secure_coding]
-      sprintf(&z[n], "?%s", fZQuery);
-      n += strlen(&z[n]);
-   }
-   if (fZFragment) {
-      // coverity[secure_coding]
-      sprintf(&z[n], "#%s", fZFragment);
-   } else {
-      z[n] = 0;
-   }
+         !(fZPath && fZPath[0] == '/'))
+      uri.Append("/");
+
+   if (fZPath)
+      uri.Append(fZPath);
+
+   if (fZQuery)
+      uri.Append(Form("?%s", fZQuery));
```

Comment:
Can you use `TString::Format()` instead of `Form` (the latter uses a static buffer)

Reply:
Done